EasyReflectometry/calculators/refl1d/wrapper.py

```diff
@@ -218,26 +190,82 @@ def sld_profile(self, model_name: str) -> Tuple[np.ndarray, np.ndarray]:
         :param model_name: the model name
         :return: z and sld(z)
         """
-        structure = model.Stack()
-        # -1 to reverse the order
-        for i in self.storage['model'][model_name]['items'][::-1]:
-            if i.repeat.value == 1:
-                # -1 to reverse the order
-                for j in range(len(i.stack))[::-1]:
-                    structure |= i.stack[j]
-            else:
-                stack = model.Stack()
-                # -1 to reverse the order
-                for j in range(len(i.stack))[::-1]:
-                    stack |= i.stack[j]
-                structure |= model.Repeat(stack, repeat=i.repeat.value)
-
-        q = names.QProbe(
-            np.linspace(0.001, 0.3, 10),
-            np.linspace(0.001, 0.3, 10),
-            intensity=self.storage['model'][model_name]['scale'],
-            background=self.storage['model'][model_name]['bkg'],
+        sample = _build_sample(self.storage, model_name)
+        probe = _get_probe(
+            q_array=np.linspace(0.001, 0.3, 10),
+            dq_array=np.linspace(0.001, 0.3, 10),  # TODO why would we use a steadily increasing dq (resolution)?
+            model_name=model_name,
+            storage=self.storage,
         )
-        z, sld, _ = names.Experiment(probe=q, sample=structure).smooth_profile()
+        z, sld, _ = names.Experiment(probe=probe, sample=sample).smooth_profile()
         # -1 to reverse the order
         return z, sld[::-1]
+
+
+def _get_oversampling_q(q_array: np.ndarray, dq_array: np.ndarray, oversampling_factor: int) -> np.ndarray:
+    argmin = np.argmin(q_array)
+    argmax = np.argmax(q_array)
+    return np.linspace(
+        q_array[argmin] - RESOLUTION_PADDING * dq_array[argmin],  # get dq (resolution) corresponding to the smallest q
+        q_array[argmax] + RESOLUTION_PADDING * dq_array[argmax],  # get dq (resolution) corresponding to the largest q
+        oversampling_factor * len(q_array),
```

Comment:
Why find out the index first in order to get the min/max?
How about 
```python
q_array[argmin] - RESOLUTION_PADDING * np.amin(dq_array)
q_array[argmin] + RESOLUTION_PADDING * np.amax(dq_array)
```

Reply:
Careful, from the dq array we are fetching the dq value corresponding to the largest element in the q array.